Reply/app/src/main/java/com/example/reply/ui/navigation/ReplyNavigationActions.kt

```diff
@@ -69,4 +76,7 @@ class ReplyNavigationActions(navController: NavHostController) {
             restoreState = true
         }
     }
+    val navigateToDetail: (emailId: Long) -> Unit = { emailId ->
+        navController.navigate("${ReplyRouts.INBOX_ROUTE}/$emailId") {}
```

Comment:
Because we're explicitly navigating to the detail screen while on a small screen, this ends up acting a bit strange to the user.

If I'm on a small screen (say the outside screen of a foldable) and I'm on the inbox list, let's say I click on a conversation. That will navigate me to that specific conversation.

If I then unfold my device, I will still be seeing the conversation take up the entire screen, even though I would have room for a list/detail view.

If I then navigate back, I will then see the list and detail together (and probably the detail I just closed)

This will also be the case when on a tablet and entering/leaving multi-window mode, or resizing on Chrome OS.

Ideally, we wouldn't be navigating at all to the detail (at least, not calling `navController.navigate`) and then handling "navigation" (as perceived by the user) at the route capable of displaying the list, detail, or both.

Reply:
Yes, I am curious on this case too, I am not sure what should be guidance here. May be @simona-anomis can share insights too. If user is deeply navigated to nested screen, and unfold the device->
1:) We pop navigation and re-render detail-list set
2:) We Let detail screen expand
3:) Don't use navigate for Mobile if we suspect user can expand/unfold screen for this purpose/

Reply:
The conclusion we came to here for publishing https://developer.android.com/guide/topics/large-screens/navigation-for-responsive-uis#responsive_content_destinations was always to do 3, and assume the user can always resize the screen.

Navigating as a side-effect of an window size changing makes it a lot more difficult to save user state, and also can be frustrating when reversing a size change. (user rotates the device accidentally, rotates back, and their state or location in the app is gone).

What that roughly would look like here is the email route can display the list, detail, or both, and then it handles "navigation" internally, with a `BackHandler` when just the detail is showing, that updates the state driving which one to show if there isn't enough space available.

Reply:
I think I agree with that too, I will then keep it as top level navigation and handle using BackHandler assuming user can resize screen anytime.